### PR TITLE
Added nginx.conf and modified docker-compose.yml

### PR DESCRIPTION
**Pull request by ShashankBhake**
## Summary by CodeRabbit

- **New Features**
  - Added a new Nginx service to the Docker setup for improved web serving and load balancing capabilities.
  - Configured Nginx to handle requests for various cryptocurrency services, enhancing request routing and traffic management.

- **Enhancements**
  - Implemented upstream server groups in Nginx for better scalability and reliability of cryptocurrency-related services.
  - Enhanced request handling with dedicated location blocks for specific cryptocurrency endpoints.

## Summary by CodeRabbit

- **New Features**
	- Introduced an Nginx service to the Docker setup for enhanced request handling and load balancing for cryptocurrency services.
	- Enhanced Nginx configuration for efficient request routing to various backend services based on specific paths.

- **Improvements**
	- Optimized traffic management with multiple server blocks for improved scalability and reliability in handling high loads. 
	- Streamlined provisioning process for Nginx configuration, ensuring necessary resources are readily available.

### CODE REVIEW

File: resources/nginx/nginx.conf [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+        proxy_set_header X-Forwarded-Proto $scheme;
+
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'Origin, Content-Type, Accept, Authorization';
+    }
+}
+
+server {
```
Comment (susruth) [inline resources/nginx/nginx.conf:168]: let's make this bitcoin.localhost/electrum

File: resources/nginx/nginx.conf [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
```
Comment (susruth) [inline resources/nginx/nginx.conf:151]: make this bitcoin.localhost/electrs

File: resources/nginx/nginx.conf [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
```
Comment (susruth) [inline resources/nginx/nginx.conf:66]: make this bitcoin.localhost/zmqpubrawtx

File: resources/nginx/nginx.conf [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    listen 80;
+    server_name bitcoin_zmqpubrawblock.localhost;
```
Comment (susruth) [inline resources/nginx/nginx.conf:51]: make this bitcoin.localhost/zmqpubrawblock

File: resources/nginx/nginx.conf [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'Origin, Content-Type, Accept, Authorization';
+    }
+}
+
+server {
+    listen 80;
+    server_name bitcoin_p2p.localhost;
```
Comment (susruth) [inline resources/nginx/nginx.conf:34]: make this bitcoin.localhost/p2p

File: resources/nginx/nginx.conf [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+
+        add_header 'Access-Control-Allow-Origin' '*';
+        add_header 'Access-Control-Allow-Methods' 'GET, POST, OPTIONS';
+        add_header 'Access-Control-Allow-Headers' 'Origin, Content-Type, Accept, Authorization';
+    }
+}
+
+server {
+    listen 80;
+    server_name bitcoin_rpc.localhost;
```
Comment (susruth) [inline resources/nginx/nginx.conf:17]: make this make this bitcoin.localhost/rpc

**Review by r4reetik — APPROVED** [review]

**Pull request merged**